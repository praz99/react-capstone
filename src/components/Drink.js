import React from 'react';
import PropTypes from 'prop-types';
import '../styles/Drink.css';

const Drink = ({ drink }) => (
  <div className="drink-container" style={{ backgroundImage: `url(${drink.strDrinkThumb})` }}>
    <a href={`/details/${drink.idDrink}`} className="drink-button" data-testid="details-link">{drink.strDrink}</a>
  </div>
);

Drink.propTypes = {
  strDrinkThumb: PropTypes.string,
  srtDrink: PropTypes.string,
}.isRequired;

export default Drink;
