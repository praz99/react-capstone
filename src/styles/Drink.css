.drink-container {
  height: 50vh;
  background-size: cover;
  display: flex;
  align-items: flex-end;
  border: 3px solid red;
  border-radius: 8px;
}

.drink-button {
  background: #00000074;
  color: white;
  font-weight: 700;
  border: 2px solid red;
  border-radius: 8px;
  padding: 5px;
  margin: 10px;
  width: 100%;
}

.drink-button:hover {
  background: #000000ce;
}
