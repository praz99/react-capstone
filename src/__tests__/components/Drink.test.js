import React from 'react';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';
import Drink from '../../components/Drink';

afterEach(cleanup);

const drinkOne = {
  "idDrink":"13501",
  "strDrink":"ABC",
  "strDrinkThumb":"https:\/\/www.thecocktaildb.com\/images\/media\/drink\/tqpvqp1472668328.jpg"
};

it('renders item component correctly', () => {
  const drink = renderer.create(<Drink drink={drinkOne}/>).toJSON();
  expect(drink).toMatchSnapshot();
});

it('renders a link to see drink details', () => {
  const { getByTestId } = render(<Drink drink={drinkOne}/>);
  const link = getByTestId('details-link');
  expect(link.innerHTML).toMatch(drinkOne.strDrink); 
});
