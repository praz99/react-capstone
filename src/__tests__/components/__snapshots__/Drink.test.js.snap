// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders item component correctly 1`] = `
<div
  className="drink-container"
  style={
    Object {
      "backgroundImage": "url(https://www.thecocktaildb.com/images/media/drink/tqpvqp1472668328.jpg)",
    }
  }
>
  <a
    className="drink-button"
    data-testid="details-link"
    href="/details/13501"
  >
    ABC
  </a>
</div>
`;
